src/components/EmailValidationForm.tsx

```diff
@@ -1,71 +1,88 @@
+import { CheckSoloIcon, XIcon } from "@nypl/dgx-svg-icons";
 
 export interface EmailValidationFormState {
-  sent: boolean;
+  validated: boolean;
 }
 
 export interface EmailValidationFormOwnProps {
-  library: LibraryData;
+  email: string;
+  libraryInfo: {};
+  uuid: string;
   store: Store<State>;
-  fetchLibrary: (uuid: string) => LibraryData;
-}
-
-export interface EmailValidationFormDispatchProps {
-  validateEmail: (data: FormData) => Promise<void>;
 }
 
 export interface EmailValidationFormStateProps {
   error?: FetchErrorData;
 }
 
+export interface EmailValidationFormDispatchProps {
+  validateEmail: (data: FormData) => Promise<void>;
+}
+
 export interface EmailValidationFormProps extends EmailValidationFormOwnProps, EmailValidationFormDispatchProps, EmailValidationFormStateProps {}
 
 export class EmailValidationForm extends React.Component<EmailValidationFormProps, EmailValidationFormState> {
-
   constructor(props) {
     super(props);
-    this.sendEmail = this.sendEmail.bind(this);
-    this.state = { sent: false };
+    this.validate = this.validate.bind(this);
+    this.renderInfoText = this.renderInfoText.bind(this);
+    this.state = { validated: false };
   }
 
-  async sendEmail(data: FormData): Promise<void> {
-    await this.props.validateEmail(data);
-    await this.props.fetchLibrary(this.props.library.uuid);
-    this.setState({ sent: !this.props.error });
+  convert(email: string): string {
+    return (email.includes("_") ? email.replace("_", " ") : email.replace(" ", "_"));
   }
 
-  render(): JSX.Element {
-    let icon = this.state.sent ? <CheckSoloIcon /> : null;
-    let alreadyValidated = this.props.library.urls_and_contact.validated !== "Not validated";
-    let hasEmail = !!this.props.library.urls_and_contact.contact_email;
+  renderTitle(emailString: string, emailAddress: string, alreadyValidated: boolean): JSX.Element[] {
+    let titleString = `${emailString}${emailAddress ? ": " + emailAddress : ""}`;
+    let icon = ((alreadyValidated || this.state.validated) && !this.props.error) ? <CheckSoloIcon /> : <XIcon />;
+    return [<span>{titleString}</span>, icon];
+  }
+
+  async validate(email: string, data: FormData): Promise<void> {
+    data.append("email", email);
+    await this.props.validateEmail(data);
+    this.setState({ validated: true });
+  }
 
-    let buttonText = hasEmail ? "Validate email address" : "No email address configured";
-    let buttonContent = <span>{buttonText}{icon}</span>;
-    let infoText = (alreadyValidated && !this.state.sent && !this.props.error) ? "Already validated" : null;
-    let successText = `Successfully validated ${this.props.library.urls_and_contact.contact_email}`;
+  renderInfoText(emailString: string, emailAddress: string, validated: string, alreadyValidated: boolean): string {
+    if (!emailAddress) {
+      return `No ${emailString} configured`;
+    } else if (!alreadyValidated) {
+      return "Not validated";
```

Comment:
Do you have a screenshot of this to add to the PR just as an example?

Reply:
@adriana-alter here's an unanswered question.

Reply:
Ah.  The answer was "no."  I'll generate one--hang on...